core/src/main/java/org/mskcc/cbio/portal/scripts/ImportGeneData.java

```diff
@@ -72,119 +72,190 @@ public static void importData(File geneFile, String genomeBuild) throws IOExcept
+                        genes = new HashSet<CanonicalGene>();
+                        genesWithoutSymbolFromNomenClatureAuthority.put(geneSymbol, genes);
                     }
+                    if (refGenes==null) {
+                        refGenes = new HashSet<ReferenceGenomeGene>();
+                        refGenesWithoutSymbolFromNomenClatureAuthority.put(geneSymbol, refGenes);
+                    }
+                    genes.add(gene);
+                    refGenes.add(refGene);
+                }
+
+                if (gene!=null) {
+                    gene.setType(type);
+                }
             }
             addGenesToDB(genesWithSymbolFromNomenClatureAuthority, genesWithoutSymbolFromNomenClatureAuthority);
-            addReferenceGenomeGenesToDB(refGenesWithSymbolFromNomenClatureAuthority, 
+            addReferenceGenomeGenesToDB(refGenesWithSymbolFromNomenClatureAuthority,
                 refGenesWithoutSymbolFromNomenClatureAuthority);
         }
     }
-        
-        /**
-         * Iterate over the genes found in the given maps and try to add them to the DB.
-         * 
-         * @param genesWithSymbolFromNomenClatureAuthority: genes with official symbol
-         * @param genesWithoutSymbolFromNomenClatureAuthority: genes without official symbol (can happen, some entrez genes
-         * have no official symbol yet, but a temporary symbol).
-         * 
-         * @throws DaoException
-         */
-        private static void addGenesToDB(Map<String, Set<CanonicalGene>> genesWithSymbolFromNomenClatureAuthority,
-                  Map<String, Set<CanonicalGene>> genesWithoutSymbolFromNomenClatureAuthority) throws DaoException {
-                     
-          DaoGeneOptimized daoGene = DaoGeneOptimized.getInstance();
-          int nrExisting = 0;
+
+    /**
+     * Adds the genes parsed from the HGNC file into the database.
+     *
+     * @param geneFile HGNC File with gene information
+     * @throws IOException
+     * @throws DaoException
+     */
+    public static void importHGNCData(File geneFile, String genomeBuild) throws IOException, DaoException {
+        Map<String, Set<CanonicalGene>> canonicalGeneMap = new LinkedHashMap<>();
+        Map<String, Set<ReferenceGenomeGene>> refGeneMap = new LinkedHashMap<>();
+        try (FileReader reader = new FileReader(geneFile)) {
+            BufferedReader buf = new BufferedReader(reader);
+            String line;
+            buf.readLine(); // skip first (header) line
+            ProgressMonitor.incrementCurValue(); // for header line
+            while ((line = buf.readLine()) != null) {
+                ProgressMonitor.incrementCurValue();
+                ConsoleUtil.showProgress();
+                String parts[] = line.split("\t", -1); // include trailing empty strings
+                if (!parts[0].matches("[0-9]+")) {
+                    ProgressMonitor.logWarning("Skipping gene with invalid entrez gene id '" + parts[1] + "'");
```

Comment:
This may be a mis-formatted line without a second column (no tabs). To avoid an exception perhaps the entire line should be printed out instead of parts[1]